cinder/volume/drivers/vmware/volumeops.py

```diff
@@ -984,6 +984,20 @@ def get_summary(self, datastore):
                                         self._session.vim, datastore,
                                         'summary')
 
+    def get_datastore_properties(self, datastore):
+        """Get datastore summary.
+
+        :param datastore: Reference to the datastore
+        :return: 'summary' property of the datastore
+        """
+        return self._session.invoke_api(vim_util, 'get_object_properties',
```

Comment:
I would replace that with `get_object_properties` with `get_object_properties_dict`, which then eliminates the need for implementing `_propset_dict`

Reply:
fixed.

Reply:
It didn't seem to have made it to github. The delta in the last force-push is empty: https://github.com/sapcc/cinder/compare/4ba98eecb24294a4f069672702dac59da3dcca33..f3d35c756e19b16e89882dffe72165081031306e

Reply:
yah it was I realized that this morning.  I have pushed an updated version with the fix.